### DISCUSSION TITLE
How do I map an array of values into discriminated union options

### DISCUSSION THREAD

**Discussion by Aaronius** [colinhacks/zod#985] [Q&A] [answered]
I'd like to be able to map an array of string values into discriminated union options and use them with `z.discriminatedUnion`. Here's what I'm trying to do:

```ts
const validationSchema = z.discriminatedUnion("taxClassification", Object.values(TaxClassification).map(taxClassification => {
  return z
    .object({
      taxClassification: z.literal(taxClassification),
      exemptPayeeCode:
        taxClassificationExemptPayeeCodeSchemaMap[taxClassification],
      exemptFatcaCode:
        taxClassificationExemptFatcaCodeSchemaMap[taxClassification],
    })
});
```

I can't though because "Source provides no match for required element at position 0 in target." 

I would typically try to type my mapped array as `[ZodDiscriminatedUnionOption, ZodDiscriminatedUnionOption, ...ZodDiscriminatedUnionOption[]` before passing it into z.discriminatedUnion, but Zod doesn't seem to expose the `ZodDiscriminatedUnionOption` type. Is there a better way?

**Comment by JacobWeisenburger** [comment, marked as answer]
I have played around with this and I don't really see a good solution without changing zod's code. Even if I manually export `ZodDiscriminatedUnionOption`, I'm still running into problems. Would you like to make this into an issue? In my opinion, this is a bug. I think that you should be able to do what you are trying to do.

**Reply by Aaronius** [reply]
Sure! That would be great! Thank you.

**Reply by cawa-93** [reply]
@Aaronius Could you show how you solve this problem?

**Reply by Aaronius** [reply]
I didn't, unfortunately. @JacobWeisenburger created https://github.com/colinhacks/zod/issues/989, which was closed but not actually fixed and I can't get anyone to re-open it. However, someone else reported the same issue in https://github.com/colinhacks/zod/issues/1181.
